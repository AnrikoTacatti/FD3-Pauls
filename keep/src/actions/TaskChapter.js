
import { OPEN_FORM_TASK_ITEM_NEW } from '../stores/const.js';
export const actionopenFormNewItemNew = (data, dispatch) => {
    dispatch({ type: OPEN_FORM_TASK_ITEM_NEW, data });
}


