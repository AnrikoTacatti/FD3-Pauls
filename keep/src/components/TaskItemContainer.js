"use strict";
import React from 'react';
import { connect } from 'react-redux';
import TaskItem from './TaskItem.js';
import {
    actionopenFormNewItemEdit,
    actionremoveTaskItem,
    actionsetTaskItemStyle,
    actionsetTaskItemPin
} from '../actions/TaskItem.js';


export class TaskItemContainer extends React.PureComponent {
    constructor(props) {
        super(props);
        this.state = {
            keychapter: this.props.keychapter,
            keyitem: this.props.keyitem,
            del: false
        }
    }

    openFormNewItemEdit = () => {
        let data = { name: this.props.data.name, text: this.props.data.text, activ: true, keychapter: this.props.keychapter, keyitem: this.props.keyitem };
        actionopenFormNewItemEdit(data, this.props.dispatch);
    }
    deleteTaskItem = () => {
        if (this.state.del === true) {
            let isDelete = window.confirm("Вы действительно хотите удалить?");
            if (isDelete) {
                this.divitem.className = this.divitem.className + " del";
                let data = { keychapter: this.state.keychapter, keyitem: this.state.keyitem };
                actionremoveTaskItem(data, this.props.dispatch);
            }
            else {
                this.setState({ del: false });
            }
        }
    }

    addDel = () => {
        this.setState({ del: true });
    }

    getTime(timestamp) {
        let d = new Date(timestamp);
        let timeStampCon = d.getDate() + '/' + (d.getMonth()) + '/' + d.getFullYear() + " " + d.getHours() + ':' + (d.getMinutes() < 10 ? '0' + d.getMinutes() : d.getMinutes());
        return timeStampCon;
    }

    setColor = (style, e) => {
        let data = { style: style, keychapter: this.state.keychapter, keyitem: this.state.keyitem };
        actionsetTaskItemStyle(data, this.props.dispatch);
    }

    setPin = (ifo) => {
        let pin = true;
        if (ifo === true) pin = false;
        let data = { pin: pin, keychapter: this.state.keychapter, keyitem: this.state.keyitem };
        actionsetTaskItemPin(data, this.props.dispatch);
    }

    render() {
        // console.log("render TaskItemContainer", this.props);
        return (
            <TaskItem {...this.props} del={this.state.del} cbopenFormNewItemEdit={this.openFormNewItemEdit} cbsetPin={this.setPin}
                cbaddDel={this.addDel} cbdeleteTaskItem={this.deleteTaskItem} cbgetTime={this.getTime} cbsetColor={this.setColor} ></TaskItem>
        )
    }
}

export default connect()(TaskItemContainer);
