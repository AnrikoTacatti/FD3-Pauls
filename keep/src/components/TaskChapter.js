"use strict";
import React from 'react';
import TaskItemContainer from './TaskItemContainer.js';
import Pagination from './Pagination.js';
import { icoPlus } from '../ico/ico.js';

class TaskChapter extends React.PureComponent {
    constructor(props) {
        super(props);
    }

    forhistoryTaskListschild = () => {
        let tasklistitem = [];
        let key = this.props.cbfindkey();
        if (key) {
            var i = 0;
            for (let tasklistskeychild in this.props.TaskLists[key].itemlist) {
                i++;
                let keyitem = this.props.TaskLists[key].itemlist[tasklistskeychild].key;
                tasklistitem.push(
                    <TaskItemContainer data={this.props.TaskLists[key].itemlist[tasklistskeychild]} keychapter={key} keyitem={keyitem} index={i} key={keyitem}
                        attrdata={this.props.locationPathname === undefined ? "all" : this.props.locationPathname} />
                )

            }
        }
        return tasklistitem;
    }


    forTaskListschild = () => {
        let tasklistitem = [];
        var i = 0;
        let listtask = this.props.TaskListsItemsSort;
        listtask = listtask.sort((a, b) => b.time - a.time);

        if (this.props.currentPage !== "all") {
            let lastIndex = this.props.perpage * this.props.currentPage;
            let firstIndex = lastIndex - this.props.perpage;
            listtask = listtask.slice(firstIndex, lastIndex);
        }

        for (let i = 0; i < listtask.length; i++) {
            let keyitem = listtask[i].key;
            let keychapter = listtask[i].keychapter;
            tasklistitem.push(
                <TaskItemContainer data={listtask[i]} keychapter={keychapter} keyitem={keyitem} index={i} key={keyitem}
                    attrdata={this.props.locationPathname === undefined ? "all" : this.props.locationPathname}
                />
            )
        }
        return tasklistitem;
    }

    render() {
        /* console.log("render TaskChapter", this.props);*/
        return (
            <React.Fragment>
                <div className="task-chapter" id={this.props.locationPathname === undefined ? "all" : this.props.locationPathname}>
                    <style>
                        {` .task-item{ -webkit-animation-name: none;
                                animation-name: none;} 
                            .all{ 
                                -webkit-animation-name: slideInUpAll;
                                 animation-name: slideInUpAll;
                             }
                            `
                        }
                        {this.props.locationPathname !== undefined && `.${this.props.locationPathname !== undefined && this.props.locationPathname} { 
                        -webkit-animation-name: slideInUp;
                        animation-name: slideInUp;
                        opacity: 0; }` }
                    </style>
                    <div className="task-chapter__header">
                        <div className="task-chapter__title">
                            {Object.keys(this.props.TaskLists).length > 0 && (this.props.locationPathname === undefined ? "Все" : this.props.cbfindName())}

                        </div>
                        <div className="task-chapter__tools">
                            {Object.keys(this.props.TaskLists).length > 0 && (this.props.locationPathname !== undefined && <span className="task-chapter__tools__Plus" onClick={this.props.cbopenFormNewItemNew} >{icoPlus()} </span>)}

                        </div>
                    </div>
                    <div>{this.props.locationPathname === undefined && <Pagination perpage={this.props.perpage} total={this.props.TaskListsItemsSort.length} />}</div>
                    <div className="task-item-list">
                        {this.props.locationPathname === undefined ? this.forTaskListschild() : this.forhistoryTaskListschild()}
                    </div>
                </div>
            </React.Fragment>

        );
    }


}


export default TaskChapter;
