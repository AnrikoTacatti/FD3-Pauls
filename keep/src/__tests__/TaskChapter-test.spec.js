"use strict";
import React from 'react';
import TaskChapter from '../components/TaskChapter';
import TaskItemContainer from '../components/TaskItemContainer.js';

let taskLists = {
    "0": {
        "itemlist": {
            "-Mnu6P7dMJTeVv8VESLh": {
                "name": "Мои заметки № 1",
                "text": "Мои заметки № 1",
                "time": 1636283818894,
                "key": "-Mnu6P7dMJTeVv8VESLh"
            },
            "-Mnu6RA2cuqhZiUml2eL": {
                "name": "Мои заметки № 2",
                "pin": true,
                "text": "Мои заметки № 2",
                "time": 1636283827275,
                "key": "-Mnu6RA2cuqhZiUml2eL"
            }
        },
        "name": "Мои заметки",
        "url": "moi-zametki",
        "key": "0"
    },
    "1": {
        "itemlist": {
            "-Mnu6lwJvGFmsBIjhPBh": {
                "name": "Заметки работа №1",
                "text": "Заметки работа №1",
                "time": 1636283916422,
                "key": "-Mnu6lwJvGFmsBIjhPBh"
            },
            "-Mnu6n_j_6Szfb5fRtuR": {
                "name": "Заметки работа №2",
                "text": "Заметки работа №2",
                "time": 1636283923175,
                "key": "-Mnu6n_j_6Szfb5fRtuR"
            }
        },
        "name": "Заметки работа",
        "url": "zametki-rabota",
        "key": "1"
    },
    "-MnosmgmdFWwUg_Qyj5o": {
        "itemlist": {
            "-Mnu71_BYp5YokQlvrg4": {
                "name": "Заметки учеба №111",
                "pin": true,
                "style": {
                    "backgroundColor": "#ffecf1",
                    "border": "1px solid #f9c8ca"
                },
                "text": "Заметки учеба №1",
                "time": 1636466420836,
                "key": "-Mnu71_BYp5YokQlvrg4"
            },
            "-Mnu736Ol-_bA2P_gxQg": {
                "name": "Заметки учеба №2",
                "text": "Заметки учеба №2",
                "time": 1636283990848,
                "key": "-Mnu736Ol-_bA2P_gxQg"
            },
            "-Mnu74Jhfq6DAtIDT796": {
                "name": "Заметки учеба №3",
                "text": "",
                "time": 1636465260116,
                "key": "-Mnu74Jhfq6DAtIDT796"
            }
        },
        "name": "Заметки учеба",
        "time": 1636196167296,
        "url": "zametki-ucheba",
        "key": "-MnosmgmdFWwUg_Qyj5o"
    },
    "-Mnot8F4fnIev0gGlnxg": {
        "itemlist": {
            "-Mnu7QdYFIdU_yM1_eCE": {
                "name": "Заметки хобби № 1",
                "text": "Заметки хобби № 1",
                "time": 1636284087310,
                "key": "-Mnu7QdYFIdU_yM1_eCE"
            },
            "-Mnu7SIorvdewdoTTefa": {
                "name": "Заметки хобби № 2",
                "text": "Заметки хобби № 2",
                "time": 1636284094021,
                "key": "-Mnu7SIorvdewdoTTefa"
            }
        },
        "name": "Заметки хобби",
        "time": 1636196193514,
        "url": "zametki-hobbi",
        "key": "-Mnot8F4fnIev0gGlnxg"
    }
};



let taskListsItemsSort = [
    {
        "name": "Заметки учеба №111",
        "pin": true,
        "style": {
            "backgroundColor": "#ffecf1",
            "border": "1px solid #f9c8ca"
        },
        "text": "Заметки учеба №1",
        "time": 1636466420836,
        "key": "-Mnu71_BYp5YokQlvrg4",
        "keychapter": "-MnosmgmdFWwUg_Qyj5o"
    },
    {
        "name": "Заметки хобби № 7",
        "text": "Заметки хобби № 7",
        "time": 1636284129331,
        "key": "-Mnu7_uMv7uxZQNdvVOf",
        "keychapter": "-Mnot8F4fnIev0gGlnxg"
    },
    {
        "name": "Заметки хобби № 1",
        "text": "Заметки хобби № 1",
        "time": 1636284087310,
        "key": "-Mnu7QdYFIdU_yM1_eCE",
        "keychapter": "-Mnot8F4fnIev0gGlnxg"
    },
    {
        "name": "Заметки учеба №11",
        "style": "background-color: rgb(232, 250, 252); border: 1px solid rgb(178, 217, 236);",
        "text": "Заметки учеба №11",
        "time": 1636284065035,
        "key": "-Mnu7FEYa0Vr8GBoy_Mc",
        "keychapter": "-MnosmgmdFWwUg_Qyj5o"
    },
    {
        "name": "Заметки работа №2",
        "text": "Заметки работа №2",
        "time": 1636283923175,
        "key": "-Mnu6n_j_6Szfb5fRtuR",
        "keychapter": "1"
    },
    {
        "name": "Заметки работа №1",
        "text": "Заметки работа №1",
        "time": 1636283916422,
        "key": "-Mnu6lwJvGFmsBIjhPBh",
        "keychapter": "1"
    },
    {
        "name": "Мои заметки № 10",
        "text": "Мои заметки № 10",
        "time": 1636283900074,
        "key": "-Mnu6hxwC6QMIoX2YcMT",
        "keychapter": "0"
    },
    {
        "name": "Мои заметки № 8",
        "text": "Мои заметки № 8",
        "time": 1636283876009,
        "key": "-Mnu6c5GJtYyhTBYAQ9d",
        "keychapter": "0"
    },

];

describe('TaskChapter-test', () => {
    let openFormNewItemNew = jest.fn();
    let findName = jest.fn(() => "Заметки учеба");
    let findkey = jest.fn(() => "-MnosmgmdFWwUg_Qyj5o");
    let state = {
        TaskLists: taskLists,
        locationPathname: "zametki-ucheba",
        TaskListsItemsSort: taskListsItemsSort,
        perpage: 5,
        currentPage: 1,
    }
    const component = shallow(
        <TaskChapter {...state} cbopenFormNewItemNew={openFormNewItemNew} cbfindName={findName} cbfindkey={findkey} />
    );

    test('TaskChapter Snapshot zametki-ucheba', () => {
        expect(component).toMatchSnapshot();
    });
    test('TaskChapter Snapshot task-chapter__title', () => {
        const title = component.find(".task-chapter__title").text();
        expect(title).toBe("Заметки учеба");
    });
    test('TaskChapter  TaskItemContainer length', () => {
        const TaskItemContainerobj = component.find(TaskItemContainer);
        expect(TaskItemContainerobj).toHaveLength(3);
    });
    test('TaskChapter task-item__tools_Trash', () => {
        const btn = component.find(".task-chapter__tools__Plus");
        btn.simulate('click');
        expect(openFormNewItemNew).toHaveBeenCalled();
    });
});
describe('TaskChapter-test all', () => {
    let openFormNewItemNew = jest.fn();
    let findName = jest.fn();
    let findkey = jest.fn();
    let state = {
        TaskLists: taskLists,
        locationPathname: undefined,
        TaskListsItemsSort: taskListsItemsSort,
        perpage: 5,
        currentPage: 1,
    }
    const component = shallow(
        <TaskChapter {...state} cbopenFormNewItemNew={openFormNewItemNew} cbfindName={findName} cbfindkey={findkey} />
    );

    test('TaskChapter forTaskListschild', () => {
        const title = component.find(".task-chapter__title").text();
        expect(title).toBe("Все");
    });
    test('TaskChapter forTaskListschild length', () => {
        const TaskItemContainerobj = component.find(TaskItemContainer);
        expect(TaskItemContainerobj).toHaveLength(5);
    });
});








