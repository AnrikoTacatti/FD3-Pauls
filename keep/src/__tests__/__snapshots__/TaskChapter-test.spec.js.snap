// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskChapter-test TaskChapter Snapshot zametki-ucheba 1`] = `
<Fragment>
  <div
    className="task-chapter"
    id="zametki-ucheba"
  >
    <style>
       .task-item{ -webkit-animation-name: none;
                                animation-name: none;} 
                            .all{ 
                                -webkit-animation-name: slideInUpAll;
                                 animation-name: slideInUpAll;
                             }
                            
      .zametki-ucheba { 
                        -webkit-animation-name: slideInUp;
                        animation-name: slideInUp;
                        opacity: 0; }
    </style>
    <div
      className="task-chapter__header"
    >
      <div
        className="task-chapter__title"
      >
        Заметки учеба
      </div>
      <div
        className="task-chapter__tools"
      >
        <span
          className="task-chapter__tools__Plus"
          onClick={[MockFunction]}
        >
          <svg
            aria-hidden="true"
            data-icon="plus"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
              fill="currentColor"
            />
          </svg>
           
        </span>
      </div>
    </div>
    <div />
    <div
      className="task-item-list"
    >
      <Connect(TaskItemContainer)
        attrdata="zametki-ucheba"
        data={
          Object {
            "key": "-Mnu71_BYp5YokQlvrg4",
            "name": "Заметки учеба №111",
            "pin": true,
            "style": Object {
              "backgroundColor": "#ffecf1",
              "border": "1px solid #f9c8ca",
            },
            "text": "Заметки учеба №1",
            "time": 1636466420836,
          }
        }
        index={1}
        key="-Mnu71_BYp5YokQlvrg4"
        keychapter="-MnosmgmdFWwUg_Qyj5o"
        keyitem="-Mnu71_BYp5YokQlvrg4"
      />
      <Connect(TaskItemContainer)
        attrdata="zametki-ucheba"
        data={
          Object {
            "key": "-Mnu736Ol-_bA2P_gxQg",
            "name": "Заметки учеба №2",
            "text": "Заметки учеба №2",
            "time": 1636283990848,
          }
        }
        index={2}
        key="-Mnu736Ol-_bA2P_gxQg"
        keychapter="-MnosmgmdFWwUg_Qyj5o"
        keyitem="-Mnu736Ol-_bA2P_gxQg"
      />
      <Connect(TaskItemContainer)
        attrdata="zametki-ucheba"
        data={
          Object {
            "key": "-Mnu74Jhfq6DAtIDT796",
            "name": "Заметки учеба №3",
            "text": "",
            "time": 1636465260116,
          }
        }
        index={3}
        key="-Mnu74Jhfq6DAtIDT796"
        keychapter="-MnosmgmdFWwUg_Qyj5o"
        keyitem="-Mnu74Jhfq6DAtIDT796"
      />
    </div>
  </div>
</Fragment>
`;
